Src/SwqlStudio/ObjectExplorer/TreeNodesBuilder.cs

```diff
@@ -0,0 +1,197 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Windows.Forms;
+using SwqlStudio.Metadata;
+
+namespace SwqlStudio
+{
+    internal class TreeNodesBuilder
+    {
+        public EntityGroupingMode EntityGroupingMode { get; set; }
+
+        private static TreeNodeWithConnectionInfo MakeEntityTreeNode(IMetadataProvider provider, Entity entity)
+        {
+            var entityNode = CreateEntityNode(provider, entity);
+
+            // Add keys
+            AddPropertiesToNode(provider, entityNode, entity.Properties.Where(c => c.IsKey));
+
+            // Add the simple Properties
+            AddPropertiesToNode(provider, entityNode, entity.Properties.Where(c => !c.IsInherited && !c.IsNavigable && !c.IsKey));
+
+            // Add the inherited Properties
+            AddPropertiesToNode(provider, entityNode, entity.Properties.Where(c => c.IsInherited && !c.IsNavigable && !c.IsKey));
+
+            // Add the Navigation Properties
+            AddPropertiesToNode(provider, entityNode, entity.Properties.Where(c => c.IsNavigable));
+
+            AddVerbsToNode(entityNode, entity, provider);
+            return entityNode;
+        }
+
+        private static void AddVerbsToNode(TreeNode entityNode, Entity entity, IMetadataProvider provider)
+        {
+            foreach (var verb in entity.Verbs.OrderBy(v => v.Name))
+            {
+                TreeNode verbNode = CreateNode(provider, verb.Name, ImageKeys.Verb, verb);
+                verbNode.ToolTipText = DocumentationBuilder.ToToolTip(verb);
+
+                var argumentsPlaceholder = new ArgumentsPlaceholderTreeNode(verb, provider);
+                verbNode.Nodes.Add(argumentsPlaceholder);
+
+                entityNode.Nodes.Add(verbNode);
+            }
+        }
+
+        private static void AddPropertiesToNode(IMetadataProvider provider, TreeNode entityNode, IEnumerable<Property> properties)
+        {
+            foreach (Property property in properties.OrderBy(c => c.Name))
+            {
+                string name = DocumentationBuilder.ToNodeText(property);
+                var imageKey = ImageKeys.GetImageKey(property);
+                TreeNode node = CreateNode(provider, name, imageKey, property);
+                node.ToolTipText = DocumentationBuilder.ToToolTip(property);
+                entityNode.Nodes.Add(node);
+            }
+        }
+
+        public static void RebuildVerbArguments(TreeNode verbNode, IMetadataProvider provider)
```

Comment:
Could this be handled in `AddVerbsToNode` instead?

Reply:
No, it was designed as lazyloading